### ISSUE TITLE
Custom components are not visible after the second render caused by React StrictMode

### ISSUE THREAD

**Issue by bulaj** [handsontable/handsontable#9723]
New behaviour of StrictMode component introduced by React in version 18 causes react components to simulate their unmounting and remounting after first mount in developer mode (https://reactjs.org/docs/strict-mode.html#ensuring-reusable-state). This is meant to ensure that effects are properly cleaned in case of multiple mounting/destroying components.

The above causes custom rendered components in handsontable to disappear after the simulated unmounting and remounting of handsontable wrapper.
Notes:
- This behaviour occurs only in developer mode and when handsontable is under `StrictMode` component in components tree:
```jsx
<React.StrictMode>
    <SomeComponents />
    <Handsontable />
</React.StrictMode>
```
Usually, StrictMode is used to wrap whole application in it, so if we want to avoid double-mounting some workarounds are needed:
```jsx
<React.Fragment>
    <React.StrictMode>
        <SomeComponents />
    </React.StrictMode>
    <Handsontable />
</React.Fragment>
```
- Reacts StrictMode is used by default in `create-react-app`, and is quite heavily advised by its developer team.

**Comment by jansiegel** [comment]
@bulaj Could you help me with reproducing the problem? I created a fresh app with `create-react-app` and implemented the `HotTable` component with a custom, component-based renderer wrapped with `React.StrictMode` (which wasn't there by default btw 🤔):

```jsx
    <React.StrictMode>
      <HotTable
        data={[
          [1, 2, 3],
          [1, 2, 3]
        ]}
        colWidths={800}
        rowHeaders={true}
        colHeaders={true}
        licenseKey="non-commercial-and-evaluation"
      >
        <RendererComponent hot-renderer/>
      </HotTable>
    </React.StrictMode>
```

And the renderer components seem to render fine. 
There is a nasty problem with them disappearing when scrolling (especially sideways), but it doesn't seem to be exclusive to `React.StrictMode` and it also happens on your PR.

What am I missing?

quick demo on codesandbox: https://codesandbox.io/s/gallant-grothendieck-2w14jr?file=/src/App.js

**Comment by bulaj** [comment]
@jansiegel the problem is still persistent on master branch of component-comparison repo. Just change data loading to default state instead of in useEffect and change default page to all_grids in Layout.tsx.

**Comment by AMBudnik** [comment]
This PR can be closed in the favor of https://github.com/handsontable/handsontable/pull/10263, released in Handsontable v12.3.3
